### PR TITLE
fix scrappy and mindseye working as passive ability

### PR DESCRIPTION
**Pull request by contra1337**
- change the attribute apply to more properly use the cancelled param
- apply the attribute for active and passive abilities
- mark minds eye as partially implemented

### CODE REVIEW

File: src/field/pokemon.ts [hunk 1]
```diff
@@ -901,14 +901,16 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
       return this.isTerastallized() ? 2 : 1;
     const types = this.getTypes(true, true);
 
-    const ignorableImmunities = source?.getAbility()?.getAttrs(IgnoreTypeImmunityAbAttr) || [];
-    const cancelled = new Utils.BooleanHolder(false);
+    let multiplier = types.map(defType => {
+      if (source && source.hasAbilityWithAttr(IgnoreTypeImmunityAbAttr)) {
```
Comment (contra1337) [inline src/field/pokemon.ts:905]: i think hasAbilityWithAttr might be redundant with how applyAbAttrs works, can remove if desired but was suggested to use it

File: src/data/ability.ts [hunk 2]
```diff
@@ -2483,11 +2483,11 @@ export class IgnoreTypeImmunityAbAttr extends AbAttr {
   }
 
   apply(pokemon: Pokemon, passive: boolean, cancelled: Utils.BooleanHolder, args: any[]): boolean {
-    if (this.defenderType !== (args[1] as Type)) {
-      return false;
+    if (this.defenderType === (args[1] as Type) && this.allowedMoveTypes.some(type => type === (args[0] as Type))) {
```
Comment (Xavion3) [inline src/data/ability.ts:2486]: This could just be an includes right, doesn't have to be a some?

File: src/field/pokemon.ts (hunk 1 above)
Comment (Xavion3) [reply src/field/pokemon.ts:905]: Yeah, the apply handles this so this is redundant.

**Review by Xavion3 — CHANGES_REQUESTED** [review]

File: src/data/ability.ts (hunk 2 above)
Comment (contra1337) [reply src/data/ability.ts]: yep good point thanks

**Review by Xavion3 — APPROVED** [review]

**Review by Madmadness65 — APPROVED** [review]

**Pull request merged**